### PR TITLE
Add label smoothing in functor

### PR DESCRIPTION
**Pull request by marigoold**
背景：https://github.com/Oneflow-Inc/oneflow/issues/8857

概述：为 `CrossEntropyLoss` 增加 `label_smoothing` 参数

实现方法：新写了一个 Functor：CrossEntropyLabelSmoothing，防止和原来的 CrossEntropy 写在一起，降低可读性。新写的 Functor 不导出 Python 接口，在原来的接口里调用，如果 label_smoothing > 0 ，就调用 CrossEntropyLabelSmoothing Functor。

torch 实现：https://github.com/pytorch/pytorch/blob/df98c529480b2ece3809b19fc850f57d2054605a/aten/src/ATen/native/LossNLL.cpp#L528-L578

注意事项：
- 1.10 的 torch 的 cross_entropy 在同时输入 label_smoothing 和 ignore_index 的情况下似乎有 bug，可见 https://github.com/pytorch/pytorch/issues 的 73205 issue。在这种 case 下，oneflow 的输出一直无法和 torch 对齐，而其他 case 可以。但是验证 1.12 可以对齐，所以在单测中加了个对 torch version 的判断，如果低于 1.12 ，则避开某些 case

对不齐复现代码如下：

1.12 以后的 torch 版本
```python
In [22]: import oneflow as flow
    ...: logit = flow.randn(3, 2)
    ...: target = flow.randint(0, 2, (3, ))
    ...: weight = flow.randn(2)

In [23]: flow.nn.functional.cross_entropy(logit, target, label_smoothing=0.1, weight=weight, ignore_index=target[0].item())
Out[23]: tensor(0.4452, dtype=oneflow.float32)

In [24]: import torch; print(torch.__version__)
1.13.0a0+git684a404

In [25]: torch.nn.functional.cross_entropy(torch.tensor(logit.to_numpy()), torch.tensor(target.to_numpy()), label_smoothing=0.1, ignore_index=target[0].item(), weight=torch.tensor(weight.numpy()))
Out[25]: tensor(0.4452)
```

1.12 以前的 torch 版本
```python
In [1]: import oneflow as flow
   ...: logit = flow.randn(3, 2)
   ...: target = flow.randint(0, 2, (3, ))
   ...: weight = flow.randn(2)

In [2]: flow.nn.functional.cross_entropy(logit, target, label_smoothing=0.1, weight=weight, ignore_index=target[0].item())
Out[2]: tensor(1.7491, dtype=oneflow.float32)

In [3]: import torch; print(torch.__version__)
1.10.0+cu102

In [4]: torch.nn.functional.cross_entropy(torch.tensor(logit.to_numpy()), torch.tensor(target.to_numpy()), label_smoothing=0.1, ignore_index=target[0].item(), weight=torch.tensor(weight.numpy()))
Out[4]: tensor(1.7203)
```

---

由于 Github 对公式的支持很不友好，我用其他编辑器写了个简单推导的文档截图放在这里：

image

### CODE REVIEW

File: python/oneflow/nn/modules/loss.py [hunk 1]
```diff
@@ -158,13 +174,20 @@ def __init__(
         weight: Optional[Tensor] = None,
         ignore_index: int = -100,
         reduction: str = "mean",
+        label_smoothing: float = 0.0,
     ) -> None:
         super(CrossEntropyLoss, self).__init__(weight, reduction)
         self.ignore_index = ignore_index
+        self.label_smoothing = label_smoothing
```
Comment (MARD1NO) [inline python/oneflow/nn/modules/loss.py:181]: 在这里加个check吧

**Review by MARD1NO — APPROVED** [review]

**Review by xiezipeng-ML — APPROVED** [review]
